packages/type-system/rust/src/ontology/uri/native.rs

```
@@ -0,0 +1,34 @@
+use url::Url;
+
+use crate::ontology::uri::{error::ParseBaseUriError, BaseUri, VersionedUri};
+
+impl BaseUri {
+    pub fn new(uri: &str) -> Result<BaseUri, ParseBaseUriError> {
+        // TODO: Propagate more useful errors
```

See comment above about swallowing [url errors](https://docs.rs/url/latest/url/enum.ParseError.html)